static/js/src/support-chart.js

```diff
@@ -0,0 +1,550 @@
+}
+
+/**
+ *
+ * @param {String} chartSelector
+ * @param {Object} taskStatus
+ *
+ * Builds key for supplied chart based on task status
+ */
+function buildChartKey(chartSelector, taskStatus) {
+  var taskStatusKeys = Object.keys(taskStatus);
+  var rectDimensions = 25;
+  var rowHeight = rectDimensions + 5;
+  var gapBetweenRectAndText = 10;
+  var verticalTextAlignmentOffset = 16;
+
+  var chartKey = d3
+    .select(chartSelector)
+    .append("svg")
+    .attr("class", "chart-key")
+    .attr("width", "550")
+    .attr("height", rowHeight * taskStatusKeys.length);
+
+  taskStatusKeys.forEach(function (key, i) {
+    var keyRow = chartKey
+      .append("g")
+      .attr("class", "chart-key__row")
+      .attr("transform", "translate(0, " + rowHeight * i + ")")
+      .attr("height", rectDimensions);
+
+    keyRow
+      .append("rect")
+      .attr("class", taskStatus[key])
+      .attr("width", rectDimensions)
+      .attr("height", rectDimensions)
+      .attr("y", 0);
+
+    keyRow
+      .append("text")
+      .text(formatKeyLabel(key))
+      .attr("class", "chart-key__label")
+      .attr("x", rectDimensions + gapBetweenRectAndText)
+      .attr("y", verticalTextAlignmentOffset);
+  });
+}
+
+/**
+ *
+ * @param {String} key
+ *
+ * Formats key into readable string
+ */
+function formatKeyLabel(key) {
+  var keyLowerCase = key.toLowerCase().replace(/_/g, " ");
+  var formattedKey =
+    keyLowerCase.charAt(0).toUpperCase() + keyLowerCase.substr(1);
+  formattedKey = formattedKey.replace(
+    "Lts",
+    "Ubuntu LTS release Standard Support"
+  );
+  formattedKey = formattedKey.replace(" openstack ", " OpenStack ");
+  formattedKey = formattedKey.replace("kub", "Kub");
+  formattedKey = formattedKey.replace(
+    "Interim release",
+    "Interim release standard security maintenance (9 months)"
+  );
+  formattedKey = formattedKey.replace(
+    "Esm",
+    "LTS Expanded Security Maintenance (ESM) for Ubuntu Main (additional 5 years)"
+  );
+  formattedKey = formattedKey.replace("Cve", "CVE/Critical fixes only");
+  formattedKey = formattedKey.replace("Early", "Early preview");
+  formattedKey = formattedKey.replace(
+    "Hardware and maintenance updates",
+    "LTS standard security maintenance for Ubuntu Main (initial 5 years)"
+  );
+  formattedKey = formattedKey.replace(
+    "Main universe",
+    "LTS Expanded Security Maintenance (ESM) for Ubuntu Universe (10 years)"
+  );
+  formattedKey = formattedKey.replace(
+    "Microstack esm",
+    "Expanded Security Maintenance (ESM)"
+  );
+  formattedKey = formattedKey.replace(
+    "Pro support",
+    "Ubuntu Pro + Support coverage"
+  );
+  return formattedKey;
+}
+
+/**
+ *
+ * @param {Array} taskTypes
+ *
+ * Calculate the longest Y-Axis label
+ */
+function calculateYAxisWidth(YAxisLabels) {
+  var YAxisLabelsCopy = YAxisLabels.slice();
+  var longestLabel = YAxisLabelsCopy.sort(function (a, b) {
+    return b.length - a.length;
+  })[0];
+  return longestLabel.length * 7;
```

Comment:
7 is the `YAxisLabels.length` right?

Reply:
Essentially. The number itself is somewhat arbitrary though

Reply:
Should it be hardcoded?